### ISSUE TITLE
Import files into Tauon

### ISSUE THREAD

**Issue by maudel15** [Taiko2k/TauonMusicBox#309]
Excellent and current software that linux undoubtedly needed. However, one limitation that should be noted is the absence of an equally effective management of the physical library rather than the playlists. The import of the tracks is cumbersome: in my case I have the library positioned on a second internal disk and no attempted modification of the configuration file allows you to permanently store the downloaded files in that path. The only effective but questionable solution is to manually intervene through a file manager. Implementing this opportunity would make Tauon the best music manager for linux.

**Comment by Taiko2k** [comment]
Sorry, I'm not sure I understand.

> I have the library positioned on a second internal disk
Do you mean the issue is you cannot import from this particular location? If so, how did you install Tauon; AUR, Snap, Flatpak?

Or, do you mean the playlist doesn't update when you add new files to the disk? If so, do you know about the "Rescan Folders" function in the tab context menu?

Otherwise, perhaps you could rephrase your explanation of the issue?

**Comment by maudel15** [comment]
The rescan works perfectly and the playlists work just as well. What I was wondering is why Tauon does not provide for the import and physical organization of the audio files in a specific folder. The only solution I have found is to manually transfer folders using the file manager.

**Comment by Taiko2k** [comment]
You mean import by typing a path or a file/folder picker so that needing to use a file manager isn't nessasary?

The reason is I always thought drag and drop from file manager is easiest. Ususaly you have all your music in one big folder.

**Comment by Taiko2k** [comment]
As for the download monitor, are you sure the download actually finished when it says still on progress?

If the icon dissapears, it usually means it thought what you downloaded wasn't a music album. (Based on ratio of music files to other files)

**Comment by maudel15** [comment]
Dragging is practical but returns less consistent data. Let me explain: some software like Rhythmbox organize the import with unique criteria. With drag and drop there is no respect for these criteria because, while dragging an album into the author's folder, this can be named in the most varied ways, referring to the year, name or site of those who have it Realized.
As for the downloads, these are finished and are audio, so much so that, by dragging them into the folders, with the rescan they are regularly identified.

**Comment by Taiko2k** [comment]
You mean, you download files to your download folder, but then want to move them to your main library on another disk? There is actually the [Folder Relocate](https://github.com/Taiko2k/TauonMusicBox/wiki/Folder-Relocate) feature which intended for this purpose.

Though it is limited in how it works, perhaps this could be expanded on if needed.

**Comment by maudel15** [comment]
I'll probably be particularly stupid but I can't find the 'Folder Navigator Panel' unless you mean the playlist panel, nor can I find any 'Move playing folder here' entries.
I will try to apply myself more deeply.

**Comment by Taiko2k** [comment]
Right click the button in the top-left corner

**Comment by Taiko2k** [comment]
Hmm, now that I think of it, a "node level folder" wont show if all the playlists music is in a single folder... I should probably change it so "move playing folder" shows on any right click

**Comment by maudel15** [comment]
Thank you! Now I see the 'Folder Navigation Panel'. I still don't access 'Move playing folder here'

**Comment by Taiko2k** [comment]
If you import both top level folders to the same playlist, you should then see them as collapsable in the folder navigator and be able to right click them

**Comment by maudel15** [comment]
In truth all the solutions that refer to the positioning / physical movement of the files do not have a correct outcome.
The paths read only the root.
For example, my entire music library is located in:
/ Mnt / 46789356-b84d-4e80-9b4e-bb79743f53aa / Maurizio / Music /
The folder is displayed in the Folder Navigator as
/ Mnt / Maurizio /
If you move albums from the home with the indicated system (move playing folder), the folder is moved to
/ Mnt / 46789356-b84d-4e80-9b4e-bb79743f53aa / Maurizio /
And not in the folder below
/ Mnt / 46789356-b84d-4e80-9b4e-bb79743f53aa / Maurizio / Music /
A folder by author is then created which must be manually relocated to the correct path and the rescan must be performed several times before it is read.
I think Tauon is an excellent reader and manager but the section relating to the physical organization of the files is not very usable. We hope that the problem will be addressed in the future.

**Comment by Taiko2k** [comment]
Indeed, the physical organisation is very limited at the moment. This might be expanded on in the future, but I don't have any immediate plan for this.
